dipy/align/imaffine.py

```diff
@@ -993,11 +993,20 @@ def optimize(self, static, moving, transform, params0,
             If None:
                 Start from identity.
             The default is None.
+        ret_metric : boolean, optional
+            if True, it returns the parameters for measuring the
+            similarity between the images (default 'False').
+            The metric containing optimal parameters and
+            the distance between the images.
 
         Returns
         -------
         affine_map : instance of AffineMap
             the affine resulting affine transformation
+        xopt : optimal parameters
+            the optimal parameters (translation, rotation shear etc.)
```

Comment:
How many parameters are there here? In exactly what order are they returned?

Reply:
The number of parameters depends on the transformation.
eg. @2D translation -> 2 parameters, 2D Rotation -> 1 parameter, etc........

Reply:
this flat vector of parameter build the affine_map so maybe we could just add a method in the AffineMap class to get this parameter instead of returning them, but it deserves its own PR